url28f181064f_simonjupp_java_skos_api_tgz-pJ8-example_CreateSKOSExampleJ8/src/uk/ac/manchester/cs/skos/SKOSDataFactoryImpl.java

```diff
@@ -0,0 +1,347 @@
+package uk.ac.manchester.cs.skos;
+
+import org.semanticweb.owlapi.model.IRI;
+import org.semanticweb.skos.*;
+import org.semanticweb.skos.extensions.SKOSLabelRelation;
+import org.semanticweb.skos.extensions.SKOSSeeLabelRelationProperty;
+import org.semanticweb.skos.properties.*;
+import org.semanticweb.skosapibinding.SKOStoOWLConverter;
+import uk.ac.manchester.cs.owl.owlapi.OWLDataFactoryImpl;
+import uk.ac.manchester.cs.skos.properties.*;
+
+import java.net.URI;
+import java.util.*;
+
+/*
+ * Copyright (C) 2007, University of Manchester
+ *
+ * Modifications to the initial code base are copyright of their
+ * respective authors, or their employers as appropriate.  Authorship
+ * of the modifications may be determined from the ChangeLog placed at
+ * the end of this file.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+/**
+ * Author: Simon Jupp<br>
+ * Date: Mar 4, 2008<br>
+ * The University of Manchester<br>
+ * Bio-Health Informatics Group<br>
+ */
+
+public class SKOSDataFactoryImpl extends OWLDataFactoryImpl implements SKOSDataFactory {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 116 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.